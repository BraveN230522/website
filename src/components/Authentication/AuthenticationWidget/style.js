import styled from '@emotion/styled';

const AuthenticationWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;

  ${({ logged }) => logged && `justify-content: center;`}

  @media (max-width: 48em) {
    min-height: calc(100% - ${({ theme }) => theme.mobileHeaderHeight});
  }

  ${({ theme }) => theme.media.phone`
    display: block;
  `}
`;

const ScrollableContainer = styled.div`
  overflow-y: auto;
  width: 100%;
  height: fit-content;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  padding: 32px 40px;

  ${({ theme }) => theme.media.phone`
    padding: 48px 12px 172px;
  `};

  ${({ theme }) => theme.media.phone`
    display: block;
    overflow: visible;
  `}
`;

export { AuthenticationWrapper, ScrollableContainer };
