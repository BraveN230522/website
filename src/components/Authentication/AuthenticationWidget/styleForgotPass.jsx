import styled from '@emotion/styled';

export const WrapperForgot = styled.div``;

export const HeadingReset = styled.div`
  /* display: flex; */
  align-items: center;
  margin-bottom: 30px;
  span {
    font-size: 28px;
    font-weight: 500;
    cursor: pointer;
  }
`;

export const TextForgot = styled.span`
  font-size: 22px;
  font-weight: 500;
  display: flex;
  justify-content: center;
`;

export const TextTitle = styled.p`
  text-align: center;
  font-size: 20px;
  font-weight: 400;
  margin-bottom: 50px;
`;

export const TextBackToLogin = styled.p`
  margin-top: 28px;
  color: #2891ff;
  font-weight: 400;
  text-align: center;
  text-decoration: underline;
  cursor: pointer;
`;
