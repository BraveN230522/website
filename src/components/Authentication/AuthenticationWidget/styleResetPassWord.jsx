import styled from '@emotion/styled';

export const Separate = styled.div`
  height: 1px;
  margin: 10px 0;
`;

export const ImgEye = styled.img`
  position: absolute;
  right: 24px;
  bottom: 23px;
  cursor: pointer;
`;

export const WrapperInput = styled.div`
  position: relative;
`;
