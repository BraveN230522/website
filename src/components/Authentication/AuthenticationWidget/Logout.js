import React, { useState } from 'react';

import { SubmitWrapper, SubmitButton, UserName, WrapperButton } from '../FormAuthentication/style';

const Logout = ({ username, onLogoutClick }) => {
  const [confirmLogout, setConfirmLogout] = useState(true);

  return (
    <SubmitWrapper>
      {confirmLogout ? (
        <UserName>You are logged in as {username}!</UserName>
      ) : (
        <UserName>Do you want to log out?</UserName>
      )}
      {confirmLogout ? (
        <SubmitButton variant="primary" onClick={() => setConfirmLogout(false)}>
          Logout
        </SubmitButton>
      ) : (
        <WrapperButton>
          <SubmitButton variant="primary" onClick={onLogoutClick}>
            Confirm
          </SubmitButton>
          <SubmitButton variant="primary" onClick={() => setConfirmLogout(true)}>
            Cancel
          </SubmitButton>
        </WrapperButton>
      )}
    </SubmitWrapper>
  );
};

export default Logout;
