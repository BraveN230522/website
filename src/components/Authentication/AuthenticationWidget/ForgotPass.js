import React, { useState, useCallback } from 'react';
import { useForm } from 'react-hook-form';
import { FormSection, FormWrapper, Input, InputLayout, SubmitButton, SubmitWrapper } from '../FormAuthentication/style';
import { HeadingReset, TextBackToLogin, TextForgot, TextTitle, WrapperForgot } from './styleForgotPass';

const ForgotPass = ({ onClick }) => {
  const [serverError, setServerError] = useState(null);
  const onChange = useCallback(() => {
    setServerError(null);
  }, []);
  const { handleSubmit, errors, watch } = useForm();
  return (
    <WrapperForgot>
      <HeadingReset>
        <span onClick={onClick}>â€¹</span>
        <TextForgot>Forgot Password</TextForgot>
      </HeadingReset>

      <TextTitle>Enter your Email and link reset password will be sent to you!</TextTitle>

      <FormSection data-testid="form-authentication-component">
        <FormWrapper>
          <InputLayout>
            <Input
              name="email"
              placeholder="Email Address"
              type="text"
              error={errors.email && 'Provide a valid Email Address.'}
              ariaLabel="email-address-input"
              onChange={onChange}
            />
          </InputLayout>
          <SubmitWrapper>
            <SubmitButton variant="primary" type="submit">
              Send request
            </SubmitButton>
          </SubmitWrapper>
        </FormWrapper>
      </FormSection>

      <TextBackToLogin onClick={onClick}>Back to Login</TextBackToLogin>
    </WrapperForgot>
  );
};

export default ForgotPass;
