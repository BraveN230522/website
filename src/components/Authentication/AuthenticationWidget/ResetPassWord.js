import React, { useState, useCallback } from 'react';
import { useForm } from 'react-hook-form';
import eyeLogoOff from '@public/svg/eye-off.svg';
import eyeLogo from '@public/svg/eye.svg';
import { FormSection, FormWrapper, Input, InputLayout, SubmitButton, SubmitWrapper } from '../FormAuthentication/style';
import { HeadingReset, TextForgot, WrapperForgot } from './styleForgotPass';
import { ImgEye, Separate, WrapperInput } from './styleResetPassWord';

const ResetPassWord = () => {
  const [serverError, setServerError] = useState(null);
  const [hidePassword, setHidePassword] = useState(true);
  const [confirmPassword, setConfirmPassword] = useState(true);
  const onChange = useCallback(() => {
    setServerError(null);
  }, []);
  const { handleSubmit, errors } = useForm();
  return (
    <WrapperForgot>
      <HeadingReset>
        <TextForgot style={{ margin: '50px 0' }}>Reset Password</TextForgot>
      </HeadingReset>

      <FormSection data-testid="form-authentication-component">
        <FormWrapper>
          <InputLayout>
            <WrapperInput>
              <Input
                name="email"
                placeholder="New Password"
                type={hidePassword ? 'password' : 'text'}
                error={errors.email && 'Provide a valid Email Address.'}
                ariaLabel="email-address-input"
                onChange={onChange}
              />
              {!hidePassword && (
                <ImgEye src={eyeLogo} alt="eye" className="iconEye" onClick={() => setHidePassword(!hidePassword)} />
              )}
              {hidePassword && (
                <ImgEye src={eyeLogoOff} alt="eye" className="iconEye" onClick={() => setHidePassword(!hidePassword)} />
              )}
            </WrapperInput>
            <Separate></Separate>
            <WrapperInput>
              <Input
                name="email"
                placeholder="Confirm Password"
                type={confirmPassword ? 'password' : 'text'}
                error={errors.email && 'Provide a valid Email Address.'}
                ariaLabel="email-address-input"
                onChange={onChange}
              />
              {!confirmPassword && (
                <ImgEye
                  src={eyeLogo}
                  alt="eye"
                  className="iconEye"
                  onClick={() => setConfirmPassword(!confirmPassword)}
                />
              )}
              {confirmPassword && (
                <ImgEye
                  src={eyeLogoOff}
                  alt="eye"
                  className="iconEye"
                  onClick={() => setConfirmPassword(!confirmPassword)}
                />
              )}
            </WrapperInput>
          </InputLayout>
          <SubmitWrapper>
            <SubmitButton variant="primary" type="submit">
              Reset PassWord
            </SubmitButton>
          </SubmitWrapper>
        </FormWrapper>
      </FormSection>
    </WrapperForgot>
  );
};

export default ResetPassWord;
