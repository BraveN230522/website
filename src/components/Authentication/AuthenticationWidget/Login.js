import React, { useState, useEffect } from 'react';

import TabSwitcher from '@components/common/TabSwitcher';
import SocialAuthentication from '@components/Authentication/SocialAuthentication';
import FormAuthentication from '@components/Authentication/FormAuthentication';
import { ScrollableContainer } from './style';
import ForgotPass from './ForgotPass';
import ResetPassWord from './ResetPassWord';

const Login = ({ closeWidget, title }) => {
  const [active, setActive] = useState(title);
  const [forgotPass, setForgotPass] = useState(false);

  const handleChageForgotPassword = () => {
    setForgotPass(true);
  };

  const handleGoBack = () => {
    setForgotPass(false);
  };

  useEffect(() => {
    if (!active) {
      setActive('Login');
    }
  }, [active]);

  return (
    <>
      {!forgotPass && (
        <TabSwitcher
          leftLabel="Login"
          rightLabel="Register"
          active={active}
          activeCallback={setActive}
          shouldSetInitialFocus
        />
      )}

      <ScrollableContainer>
        {!forgotPass && <SocialAuthentication afterSubmit={closeWidget} />}
        {!forgotPass ? (
          <FormAuthentication
            afterSubmit={closeWidget}
            formTitle={`${active} with your email:`}
            submitButtonTitle={active}
            onClick={handleChageForgotPassword}
          />
        ) : (
          <ForgotPass
            afterSubmit={closeWidget}
            formTitle="Forgot passWord"
            submitButtonTitle={active}
            onClick={handleGoBack}
          />
          // <ResetPassWord />
        )}
      </ScrollableContainer>
    </>
  );
};

export default Login;
