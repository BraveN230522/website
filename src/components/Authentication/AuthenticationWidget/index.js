import React, { useEffect, useContext } from 'react';

import UserContext from '@utils/userContext';
import { Facebook } from '../SocialAuthentication/facebook';
import { AuthenticationWrapper, ScrollableContainer } from './style';

import Logout from './Logout';
import Login from './Login';

const AuthenticationWidget = ({ closeWidget, title }) => {
  const { state: user, dispatch: userDispatch } = useContext(UserContext);
  const isLogged = Boolean(user.id);

  const handleLogoutClick = () => userDispatch({ type: 'logout' });

  return (
    <AuthenticationWrapper logged={isLogged} data-testid="authentication-widget-component">
      {isLogged ? (
        <Logout username={user.username} onLogoutClick={handleLogoutClick} />
      ) : (
        <Login closeWidget={closeWidget} title={title} />
      )}
    </AuthenticationWrapper>
  );
};

export default AuthenticationWidget;
