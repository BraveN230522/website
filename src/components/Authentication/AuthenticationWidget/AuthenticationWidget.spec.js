import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import { matchers } from 'jest-emotion';

import AuthenticationWidget from './index';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(matchers);

afterEach(() => {
  cleanup();
});

const testId = 'authentication-widget-component';

const setup = label => {
  const utils = render(<AuthenticationWidget />, {
    wrapper: withGlobals,
  });

  if (!label) {
    return utils;
  }

  let input;
  if (typeof label === 'string') {
    input = utils.getByLabelText(label);
  } else {
    input = label.map(_ => utils.getByLabelText(_));
  }

  return {
    input,
    ...utils,
  };
};

describe('<AuthenticationWidget />', () => {
  it('Renders the Authentication unchanged', () => {
    const { asFragment, queryByTestId } = setup();
    expect(asFragment()).toMatchSnapshot();
    expect(queryByTestId(testId)).toBeTruthy();
  });

  it('Switch between login and register content', () => {
    const { getByText, getAllByText } = setup();

    fireEvent.click(getAllByText('Login')[0]);
    expect(getByText('Or Login with Email Address:')).toBeTruthy();
    expect(getAllByText('Login')[1]).toBeTruthy();

    fireEvent.click(getAllByText('Register')[0]);
    expect(getByText('Or Register with Email Address:')).toBeTruthy();
    expect(getAllByText('Register')[1]).toBeTruthy();
  });
});
