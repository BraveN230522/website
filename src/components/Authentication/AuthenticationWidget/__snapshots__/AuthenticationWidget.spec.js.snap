// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticationWidget /> Renders the Authentication unchanged 1`] = `
<DocumentFragment>
  .emotion-41 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
}

@media (max-width:48em) {
  .emotion-41 {
    min-height: calc(100% - 93px);
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 60px;
}

.emotion-4 button:nth-of-type(1) {
  border-top-left-radius: 16px;
}

.emotion-4 button:nth-of-type(2) {
  border-top-right-radius: 16px;
}

@media (max-width:48em) {
  .emotion-4 button:nth-of-type(1) {
    border-top-left-radius: 0;
  }

  .emotion-4 button:nth-of-type(2) {
    border-top-right-radius: 0;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: DINAlternateRegular,sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  background-color: #eeeeee;
  color: #9c9c9c;
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-2:focus {
  outline: none;
  box-sizing: border-box;
  border: 3px solid;
}

.emotion-2::-moz-focus-inner {
  border: 0;
}

.emotion-39 {
  overflow-y: auto;
  width: 100%;
  height: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 32px 40px;
}

@media (max-width:48em) {
  .emotion-39 {
    padding: 48px 12px;
  }
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 32px 0;
}

@media (max-width:48em) {
  .emotion-16 {
    padding: 48px 0;
  }
}

.emotion-6 {
  font-family: Inter,sans-serif font-style:normal;
  font-weight: 500;
  font-size: 16px;
  text-align: center;
  height: 16px;
  padding: 0;
  margin: 0;
  margin-bottom: 12px;
  box-sizing: border-box;
  color: #000000;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:48em) {
  .emotion-14 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .emotion-14 button:last-of-type {
    margin-bottom: 0;
  }
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  font-family: Inter,sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  text-align: center;
  padding: 0;
  margin: 0;
  margin-bottom: 10px;
  background: transparent;
  border-width: 0;
  border-color: #9c9c9c;
  border-style: solid;
  box-sizing: border-box;
  border-radius: 25px;
}

.emotion-8 > img {
  width: 24px;
  height: 24px;
  margin-right: 10px;
}

@media (max-width:48em) {
  .emotion-8 {
    border-width: 1px;
  }
}

.emotion-8:hover {
  cursor: pointer;
}

.emotion-37 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-35 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

.emotion-28 div {
  width: 100%;
}

@media (max-width:48em) {
  .emotion-28 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-wrap: nowrap;
    -ms-flex-wrap: nowrap;
    flex-wrap: nowrap;
  }

  .emotion-28 div {
    width: 100%;
  }
}

.emotion-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Inter,sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  padding: 12px 24px;
  margin: 0;
  margin-bottom: 10px;
  border: 2px solid #9c9c9c;
  box-sizing: border-box;
  border-radius: 30px;
}

@media (max-width:48em) {
  .emotion-20 {
    border-width: 1px;
    box-sizing: border-box;
    border-radius: 25px;
  }
}

.emotion-33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 24px;
}

@media (max-width:48em) {
  .emotion-33 {
    padding-top: 20%;
  }
}

.emotion-31 {
  border: solid 2px #2891ff;
  color: #ffffff;
  background: #2891ff;
  height: 45px;
  font-weight: 600;
  font-size: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
  cursor: pointer;
  font-family: Inter,sans-serif;
  border-radius: 25px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 103px;
  width: 33.33%;
}

.emotion-31:hover {
  border-color: #2891ff;
  background: #ffffff;
  color: #2891ff;
}

.emotion-31 svg {
  margin-right: 6px;
}

.emotion-31:hover path {
  stroke: #000000;
}

@media (max-width:48em) {
  .emotion-31 {
    width: 100%;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: DINAlternateRegular,sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 24px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  background-color: #ffffff;
  color: #000000;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0::-moz-focus-inner {
  border: 0;
}

<div
    class="emotion-41 emotion-42"
    data-testid="authentication-widget-component"
  >
    <div
      class="emotion-4 emotion-5"
      data-testid="tab-switcher-component"
    >
      <button
        class="emotion-0 emotion-1"
      >
        Login
      </button>
      <button
        class="emotion-2 emotion-1"
      >
        Register
      </button>
    </div>
    <div
      class="emotion-39 emotion-40"
    >
      <section
        class="emotion-16 emotion-17"
        data-testid="social-authentication-component"
      >
        <span
          class="emotion-6 emotion-7"
        >
          Continue with:
        </span>
        <div
          class="emotion-14 emotion-15"
        >
          <button
            class="emotion-8 emotion-9"
          >
            <img
              alt="continue with Facebook"
              src="mocked-file"
            />
            Facebook
          </button>
          <button
            class="emotion-8 emotion-9"
          >
            <img
              alt="continue with Google"
              src="mocked-file"
            />
            Google
          </button>
          <button
            class="emotion-8 emotion-9"
          >
            <img
              alt="continue with LinkedIn"
              src="mocked-file"
            />
            LinkedIn
          </button>
        </div>
      </section>
      <section
        class="emotion-37 emotion-38"
        data-testid="form-authentication-component"
      >
        <span
          class="emotion-6 emotion-19"
        >
          Or Login with Email Address:
        </span>
        <form
          class="emotion-35 emotion-36"
        >
          <div
            class="emotion-28 emotion-29"
          >
            <div
              class="emotion-22 emotion-23"
            >
              <input
                aria-label="email-address-input"
                class="emotion-20 emotion-21"
                name="email"
                placeholder="Email Address"
                type="text"
              />
            </div>
            <div
              class="emotion-22 emotion-23"
            >
              <input
                aria-label="password-input"
                class="emotion-20 emotion-21"
                name="password"
                placeholder="Password"
                type="password"
              />
            </div>
          </div>
          <div
            class="emotion-33 emotion-34"
          >
            <button
              class="emotion-30 emotion-31 emotion-32"
            >
              Login
            </button>
          </div>
        </form>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
